packages/gatsby-source-kontent/gatsby-node.ts

```diff
@@ -43,10 +43,17 @@ exports.sourceNodes = async (
   pluginConfig: CustomPluginOptions,
 ): Promise<void> => {
   try {
-    await kontentItemsSourceNodes(api, pluginConfig);
+    if(!_.isEmpty(api.webhookBody)){ //preview run
```

Comment:
Don't we want/need some config validation here/in the block? Do we believe the options are set correctly, we are running on the preview environment and we've got all key?

Reply:
Could be - will incorporate https://github.com/Kentico/kontent-webhook-helper-js

Reply:
Used webhooks helper 
* still need to wait for https://github.com/Kentico/kontent-webhook-helper-js/pull/1
* currently there is no way to get the header's value in refresh run - issue created https://github.com/gatsbyjs/gatsby/issues/23593

Reply:
At least I have used the types in webhook helper

Reply:
❤